src/GafferOSL/OSLImage.cpp

```diff
@@ -68,77 +68,111 @@ OSLImage::OSLImage( const std::string &name )
 
 	addChild( new Plug( "channels", Plug::In, Plug::Default & ~Plug::AcceptsInputs ) );
 	addChild( new OSLCode( "__oslCode" ) );
+	addChild( new GafferImage::Constant( "__defaultConstant" ) );
 	shaderPlug()->setInput( oslCode()->outPlug() );
+	defaultConstant()->formatPlug()->setInput( defaultFormatPlug() );
+	
 
 	channelsPlug()->childAddedSignal().connect( boost::bind( &OSLImage::channelsAdded, this, ::_1, ::_2 ) );
 	channelsPlug()->childRemovedSignal().connect( boost::bind( &OSLImage::channelsRemoved, this, ::_1, ::_2 ) );
- 
 
 	// We don't ever want to change these, so we make pass-through connections.
-	outPlug()->formatPlug()->setInput( inPlug()->formatPlug() );
-	outPlug()->dataWindowPlug()->setInput( inPlug()->dataWindowPlug() );
 	outPlug()->metadataPlug()->setInput( inPlug()->metadataPlug() );
 }
 
 OSLImage::~OSLImage()
 {
 }
 
+GafferImage::FormatPlug *OSLImage::defaultFormatPlug()
+{
+	return getChild<GafferImage::FormatPlug>( g_firstPlugIndex );
+}
+
+const GafferImage::FormatPlug *OSLImage::defaultFormatPlug() const
+{
+	return getChild<GafferImage::FormatPlug>( g_firstPlugIndex );
+}
+
 GafferScene::ShaderPlug *OSLImage::shaderPlug()
 {
-	return getChild<GafferScene::ShaderPlug>( g_firstPlugIndex );
+	return getChild<GafferScene::ShaderPlug>( g_firstPlugIndex + 1 );
 }
 
 const GafferScene::ShaderPlug *OSLImage::shaderPlug() const
 {
-	return getChild<GafferScene::ShaderPlug>( g_firstPlugIndex );
+	return getChild<GafferScene::ShaderPlug>( g_firstPlugIndex + 1 );
 }
 
 Gaffer::ObjectPlug *OSLImage::shadingPlug()
 {
-	return getChild<ObjectPlug>( g_firstPlugIndex + 1 );
+	return getChild<ObjectPlug>( g_firstPlugIndex + 2 );
 }
 
 const Gaffer::ObjectPlug *OSLImage::shadingPlug() const
 {
-	return getChild<ObjectPlug>( g_firstPlugIndex + 1 );
+	return getChild<ObjectPlug>( g_firstPlugIndex + 2 );
 }
 
 Gaffer::Plug *OSLImage::channelsPlug()
 {
-	return getChild<Gaffer::Plug>( g_firstPlugIndex + 2 );
+	return getChild<Gaffer::Plug>( g_firstPlugIndex + 3 );
 }
 
 const Gaffer::Plug *OSLImage::channelsPlug() const
 {
-	return getChild<Gaffer::Plug>( g_firstPlugIndex + 2 );
+	return getChild<Gaffer::Plug>( g_firstPlugIndex + 3 );
 }
 
 GafferOSL::OSLCode *OSLImage::oslCode()
 {
-	return getChild<GafferOSL::OSLCode>( g_firstPlugIndex + 3 );
+	return getChild<GafferOSL::OSLCode>( g_firstPlugIndex + 4 );
 }
 
 const GafferOSL::OSLCode *OSLImage::oslCode() const
 {
-	return getChild<GafferOSL::OSLCode>( g_firstPlugIndex + 3 );
+	return getChild<GafferOSL::OSLCode>( g_firstPlugIndex + 4 );
+}
+
+GafferImage::Constant *OSLImage::defaultConstant()
+{
+	return getChild<GafferImage::Constant>( g_firstPlugIndex + 5 );
+}
+
+const GafferImage::Constant *OSLImage::defaultConstant() const
+{
+	return getChild<GafferImage::Constant>( g_firstPlugIndex + 5 );
+}
+
+const GafferImage::ImagePlug *OSLImage::defaultedInPlug() const
+{
+	if( inPlug()->getInput() )
+	{
+		return inPlug();
+	}
+	else
+	{
+		return defaultConstant()->outPlug();
```

Comment:
ComputeNodes should pull only on their own input plugs, not a dangling plug of another node like this. We should add a `__defaultIn` plug and connect `defaultConstant()->outPlug()` into it, and then return `defaultInPlug()` here.